examples/platforms/cc2538/uart.c

```diff
@@ -54,14 +54,24 @@ static void processTransmit(void);
 static const uint8_t *sTransmitBuffer = NULL;
 static uint16_t sTransmitLength = 0;
 
+typedef struct RecvBuffer
+{
+    // The offset of the first item written to the list.
+    uint16_t mHead;
+    // The offset of the next item to be written to the list.
+    uint16_t mTail;
+} RecvBuffer;
```

Comment:
I've been thinking for sometime that core/common should have a generic RingBuffer class that has solid unit tests.